### PR TITLE
reformat HTTP API guide

### PR DESCRIPTION
**Pull request by ericvw**
This is in preparation for #113.

### CODE REVIEW

File: doc/http-api-guide.md [hunk 1]
```diff
@@ -71,11 +72,10 @@ EOF
...
 
 
-HTTP request headers
---------------------
+### HTTP request headers
 
-The client may specify the following HTTP request headers to tell the server
-what sorts of responses it can use:
+The client may specify the following HTTP request headers to inform the server
+of its capabilities in handling the respose:
```
Comment (yupinghu) [inline doc/http-api-guide.md]: s/respose/reponse/

File: doc/http-api-guide.md [hunk 2]
```diff
@@ -29,10 +29,11 @@ available in the future.
 [Developer's Guide]: http://www.bloomberglabs.com/files/2014/07/blpapi-developers-guide-2.54.pdf
 
 
-Requesting data
----------------
+Request/Response data
```
Comment (yupinghu) [inline doc/http-api-guide.md]: Should this use ### to be consistent with the following sections?

Comment (ericvw) [inline doc/http-api-guide.md]: Nope - this is a 2nd-level heading.  The remaining headings fall under request/response.

**Comment by ericvw** [comment]
Updated PR based on typo feedback.

**Comment by ericvw** [comment]
These typos make me wonder if we could hook up aspell checking with an accepted dictionary list...

File: doc/http-api-guide.md (hunk 2 above)
Comment (yupinghu) [inline doc/http-api-guide.md]: OK, perhaps a better question is: should we switch completely to # for headings, rather than mixing underlines and #? :)

**Comment by ericvw** [comment]
Updated PR with different headings.

**Pull request merged**